proposed/2022/package-scoring-v1.md

```diff
@@ -0,0 +1,222 @@
+
+For issues which can be addressed, an issue will include an empowering message regarding how one can take action to help address the issue.
+
+EX: A package might be missing a README which a developer may not be able to easily get started with the package. They might be able to then contribute a new issue or even the inclusion of the README file to the NuGet package if it's open source.
+
+**Issue Examples:**
+
+- Invalid .nuspec / .csproj
+- Missing dependency
+- Missing README
+- No example
+- Not enough public API documentation
+- Bad semver
+- Not v1+
+- No website
+- No repository
+- Unmaintained
+- No bug tracker
+- Has CVE (Critical, High, etc)
+- Deprecated license
+- Missing license
+- Non SPDX license
+- Unsafe copyright
+- Deprecated
+- Empty package
+- Malware
+- etc
+
+#### Popularity
+
+Popularity is the more unique of the other three categories which are more issue based. Popularity is calculated as how many projects depend on a package over the past 30 days. This is scored as a percentile up to 100% (most used) to 0% (least used).
+
+Since this is not possible with current data pipelines, we will work on providing such an experience in the future and use two proxy values such as total weekly downloads and total count of packages depending on the package to be scored up to a percentile of up to 100%.
+
+- Total Weekly Downloads
+- Number of Dependents
+
+Total Score - 100
+
+#### Quality
+
+Quality is the combination of:
+
+- Follow NuGet conventions
+  - Provides a valid .csproj / .nuspec
+  - Provides a valid README
+- Provide documentation
+  - Provides an example
+  - 20% or more of the public API has xml doc comments
+- Platform support
+  - Supports all possible modern platforms i.e. [.NET 5+/.NET Standard](https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#net-5-and-net-standard) (iOS, Android, Windows, MacOS, Linux, etc)
+- Pass static analysis
+  - Code has no errors, warnings, lints, or formatting issues.
+- Support up-to-date dependencies
+  - All of the package dependencies are supported in the latest version
+  - Package supports the latest stable .NET SDK
+
+Each of these issues would be associated an arbitrary amount of points to make a total of 100. This definition of quality is debatable and thus we will create a minimal set of quality practices.
+
+Total Score - 100
+
+#### Maintenance
+
+Maintenance is the simplest of them all as it pertains to whether the package has been updated in a reasonable timeframe. Packages that have not been updated in more than a year may be unmaintained and any problems with the package may go unaddressed.
+
+- Package updated in the last year.
+
+The score may also be shown as a decayed value over the year approaching the year mark where it goes stale.
```

Comment:
I wonder how a package owner can attest "this package is still maintained but it hasn't needed an update in over 1 year". Or is this a case we don't think is common?

Reply:
I know that different people have different preferences, but I'm on the more risk adverse side. I highly value mature and stable packages. Packages with high churn increase risk of bugs, so I strongly dislike punishing (reducing the score) for "older" packages.

Consider Newtonsoft.Json for example. Excluding the recent prerelease version, the current stable version (13.0.1) is more than 12 months ago. The time between it and the previous stable version (12.0.3) was more than 12 months.  Are we really willing to say that this package is unmaintained and deserves a reduced score?

I understand the intent is to encourage packages that have bug fixes, but I think a simple "how long since the last publish date" is too simplistic and will have worse unintended consequences than the benefits it hopes to bring.

Reply:
Heard a great quote about the Rust packages (crates) today: https://youtu.be/Z3xPIYHKSoI?t=145

> They are not abandoned, they are done

I feel like this should be something to strive towards, instead of expecting packages to be constantly updated